design/Events_design.md

```diff
@@ -0,0 +1,300 @@
+# Events in Obscuro - Design document
+
+## Scope
+
+This is a design proposal for how to handle events in Obscuro.
+
+It covers two aspects:
+
+- the visibility rules for events.
+- technical implementation details.
+
+## Ethereum Events
+
+To help dApp developers design applications with a good UX, the ethereum developers invented the concept of "events" or "logs", which
+are pieces of information emitted from smart contracts, which can be streamed in real time to external applications that
+subscribe to them.
```

Comment:
I think we also need to think in terms of events as actually being written to log files. The reason we need to think this way is I'm sure tooling such as The Graph depends on reading the log files vs live streams